### PR TITLE
MULTISITE-22895: Implement --input parameter wget.

### PR DESCRIPTION
**Pull request by jonhy81**

### CODE REVIEW

File: src/TaskRunner/Commands/CloneCommands.php [hunk 1]
```diff
@@ -218,12 +219,45 @@ private function downloadChecksumFile(array $options = [
...
+     *   Name of filename to append to url.
+     *
+     * @param array $options
+     *   Command options.
+     */
+    private function buildDownloadLink($filename, array $options = [
+        'asda-url' => InputOption::VALUE_REQUIRED,
+        'asda-user' => InputOption::VALUE_REQUIRED,
+        'asda-password' => InputOption::VALUE_REQUIRED,
+        'dumpfile' => InputOption::VALUE_REQUIRED,
```
Comment (ghost) [inline src/TaskRunner/Commands/CloneCommands.php]: Option dumpfile is not used in code.

File: src/TaskRunner/Commands/CloneCommands.php [hunk 2]
```diff
@@ -218,12 +219,45 @@ private function downloadChecksumFile(array $options = [
...
+            ->option('-i', 'downloadLink.txt')
             ->option('-O', 'latest.sh1')
-            ->option('-nH', $options['asda-url'] . '/latest.sh1')
             ->option('-A', '.sh1')
             ->option('-P', './')
             ->run();
     }
+
+    /**
+     * Create file with full url to be dowloaded.
```
Comment (ghost) [inline src/TaskRunner/Commands/CloneCommands.php]: Propose to change to: `Create file for usage in wget --input-file argument in the downloadDump() function.`

File: src/TaskRunner/Commands/CloneCommands.php [hunk 3]
```diff
@@ -218,12 +219,45 @@ private function downloadChecksumFile(array $options = [
...
+     *
+     * Make use of --input file to make sure credentials are not
```
Comment (ghost) [inline src/TaskRunner/Commands/CloneCommands.php]: This comment belongs to `$this->taskExec('wget')` code in the `downloadDump()` function

File: src/TaskRunner/Commands/CloneCommands.php [hunk 4]
```diff
@@ -218,12 +219,45 @@ private function downloadChecksumFile(array $options = [
...
+     * exposed in the logs.
+     *
+     * @param string $filename
+     *   Name of filename to append to url.
+     *
+     * @param array $options
+     *   Command options.
+     */
+    private function buildDownloadLink($filename, array $options = [
```
Comment (ghost) [inline src/TaskRunner/Commands/CloneCommands.php]: Propose to change function name to be more descriptive of what it does, e.g.: `generateAsdaWgetInputFile()`.

File: src/TaskRunner/Commands/CloneCommands.php [hunk 5]
```diff
@@ -218,12 +219,45 @@ private function downloadChecksumFile(array $options = [
...
+        foreach ($disallowed as $d) {
+            if (strpos($options['asda-url'], $d) === 0) {
+                $url = str_replace($d, '', $options['asda-url']);
+            }
+        }
+
+        $downloadLink = 'https://' . $options['asda-user'] . ':' . $options['asda-password'] . '@' . $url . '/' . $filename;
+
+        $tasks[] = $this->taskFilesystemStack()
+            ->taskWriteToFile('downloadLink.txt')
```
Comment (ghost) [inline src/TaskRunner/Commands/CloneCommands.php]: Just pointing out this filename is hardcoded and thus creates a dependency to the `downloadDump()` function. Maybe better pass it as an argument to this function?

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/TaskRunner/Commands/CloneCommands.php (hunk 3 above)
Comment (jonhy81) [reply src/TaskRunner/Commands/CloneCommands.php]: Removed

File: src/TaskRunner/Commands/CloneCommands.php [hunk 6]
```diff
@@ -168,27 +173,28 @@ public function downloadDump(array $options = [
...
 
-        // Download the file.
+        // Download the .sql file.
+        $this->generateAsdaWgetInputFile($filename, $options);
         $tasks[] = $this->taskExec('wget')
-            ->option('--http-user', $options['asda-user'])
-            ->option('--http-password', $options['asda-password'])
             ->option('-O', $options['dumpfile'] . '.gz')
-            ->option('-nH', $options['asda-url'] . '/' . $filename)
+            ->option('-i', this::TEMP_INPUTFILE)
```
Comment (ghost) [inline src/TaskRunner/Commands/CloneCommands.php]: Should be `self::TEMP_INPUTFILE` I guess, since it's only used it it's own class.

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/TaskRunner/Commands/CloneCommands.php (hunk 6 above)
Comment (jonhy81) [reply src/TaskRunner/Commands/CloneCommands.php]: In fact there is a typo there. I would type. $this->TEMP_INPUTFILE instead. I would reserve :: for static members.

**Review by ghost — APPROVED** [review]

**Pull request merged**